core/trino-main/src/main/java/io/trino/util/FastutilSetHelper.java

```diff
@@ -318,4 +424,48 @@ public synchronized T get()
             }
         }
     }
+
+    private static Set<Long> createSetForLongJavaType(Set<Long> values, Type type, MethodHandle hashCodeHandle, MethodHandle equalsHandle)
+    {
+        if (values.isEmpty() || !canSkipTypeSpecificEqualsAndHashCode(type)) {
+            return new LongOpenCustomHashSet(values, 0.25f, new LongStrategy(hashCodeHandle, equalsHandle, type));
+        }
+
+        Iterator<Long> iterator = values.iterator();
+        long min = iterator.next();
+        long max = min;
+        while (iterator.hasNext()) {
+            long value = iterator.next();
+            min = min(min, value);
+            max = max(max, value);
+        }
+
+        // A Set based on a fastutil open hashset uses (next power of two of size / load-factor)
+        // slots in a hash table which is an array of longs.
+        long fastUtilSetEntries = (int) min(1 << 30, nextPowerOfTwo((long) ceil(values.size() / 0.25f)));
+        BigInteger range = BigInteger.valueOf(max)
```

Comment:
I don't get it, we are using BigInteger but then calling intValueExact. Can we use ints here as well?

Reply:
`intValueExact` is called only if `range.compareTo(BigInteger.valueOf(Integer.MAX_VALUE)) > 0` is false. It's possible for the range to exceed int and long.